apps/05Mastermind.js

```diff
@@ -29,18 +29,70 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+function generateHint(solution, guess) {
+  // Check to see if the guessed letters equal solution letters' location
+  var correctLetterLocations = 0;
+  //  Checks to see if any of the guessed letters equal the solutions letters in ANY location.
+  var correctLetters = 0;
+  // the .split splits an array by what ever character is put in-between the "" and creates a substring and returns them as an array.
+  // this var is assigned the sub string , , , , of the solution array
+  var solutionArray = solution.split("");
+  // this var is assigned the sub string , , , , of the guess array
+  var guessArray = guess.split("");
+  
+  for (var i = 0; i < solutionArray.length; i++) {
+    if (guessArray[i] === solutionArray[i]) {
+      correctLetterLocations++;
+      solutionArray[i] = null;
+      // else leave it alone.
+    }
+  }
+
+  for (var i = 0; i < solutionArray.length; i++) {
+
+    // This targetIndex is a temporary container that holds the value of i in guessArray and evaluates the .indexOf it in solutionArray. 
+    var targetIndex = guessArray.indexOf(solutionArray[i]);
+    // Switched these to test. 
+
+    // Evaluates if the container targetIndex is > -1 (meaning, it exists in the solutionArray), if so it adds a count to correctLetters and sets the value to null.
+    if (targetIndex > -1 ){
+      correctLetters++;
+      solutionArray[i] = null;
+    }
+  }
+
+    // creates a variable to hold the correct Letter and locations to be used later.
+    return correctLetterLocations + "-" + correctLetters;
 }
 
 function mastermind(guess) {
-  // your code here
+  // This is a test line to make sure my function is working. it currently equals 'abcd'
+  solution = "abcd";
```

Comment:
You can remove the predefined solution here after your tests pass.

The game will generate one for you.